### PR TITLE
Dstack: Adding feature for Blue/Green deployments

### PR DESCRIPTION
**Pull request by rdrabo**

### CODE REVIEW

File: example/example.yaml [hunk 1]
```diff
@@ -1,146 +1,40 @@
 meta:
-  division: MyDivision                           # Assigned by MyCompany
-  team_name: MyTeam
-  suite_name: yeeha
-  suite_version: 0.1.0
+  division: KTP                             # Assigned by MyCompany
```
Comment (mhktp) [inline example/example.yaml]: Should not be KTP

File: example/example.yaml [hunk 2]
```diff
@@ -1,146 +1,40 @@
...
-      log_folder: /var/log/atom           # Optional: Set the folder where your container dumps logs
-      desired_count: 1                    # Optional: How many containers should be running in total (default: 1)
-      platform: linux                     # Not Implemented Yet: (linux|windows) (default: linux)
-      env_vars:                           # Optional: Set environment variables to pass into container
+    app1:
+#      access: external|internal             # Required: If present, make the application external/internal availabile.
+      docker_image: hello-world             # Required: Docker image name
+      internal_port:                        # Required: Port(s) the application listens on inside the container. Can be an integer (single port) or array of integers (multiple ports). If routing via a url (see below), the first port in the list is used.
+       - 80
+      docker_registry: 526930246559.dkr.ecr.us-east-1.amazonaws.com # Optional (default: Docker Hub)
```
Comment (mhktp) [inline example/example.yaml]: Should be account  `0123456789012`

File: lib/dstack/options.rb [hunk 3]
```diff
@@ -30,7 +30,7 @@ def cliopts args
   Usage: dstack [command] [environment] [options]
 
   Commands:
-    deploy                   Deploys the stack
+    create                   Create the stack
```
Comment (mhktp) [inline lib/dstack/options.rb]: Breaking change. Does anything else need to be updated?

File: README.md [hunk 4]
```diff
@@ -110,6 +110,12 @@ meta:
   suite_version: 0.1.0
   suite_description: An example application
 
+global:
```
Comment (mhktp) [inline README.md]: I think that `global` should be merged under either `meta` or `all`. What is the purpose of having a 3rd top-level YAML entry?

File: README.md [hunk 5]
```diff
@@ -110,6 +110,12 @@ meta:
...
+  deployment_type:
```
Comment (mhktp) [inline README.md]: Here we would need to keep an example of the default rolling-update deployment type.

File: example/example.yaml [hunk 6]
```diff
@@ -1,146 +1,40 @@
...
         update_r53: true                    # Automatically update Route53
         #landingpage: /app/index.html       # If set, auto-redirect the root url ("http://domain/") to this
         #forcessl: true                     # Redirect all http requests to https
-        #nginx_optimizations: true          # Enable Nginx Optimizations
-      resources:                          # Optional: Set resource limits
+        nginx_optimizations: true           # Enable Nginx Optimizations
+      resources:                            # Optional: Set resource limits
         cpu: 100                            # CPU units, out of 1024 (default: 100)
         mem: 200                            # Memory required per container, in MB (default: 256)
-      health_check: tcp                   # Optional: Enables the health check in service discovery and load balancers ( tcp | http | /some/path/to/http/check ) (default: false)
```
Comment (mhktp) [inline example/example.yaml]: Why are we throwing away a perfectly good, extensive example?

File: lib/dstack/options.rb (hunk 3 above)
Comment (rdrabo) [inline lib/dstack/options.rb]: I will discuss with Jerry about this change.

File: exe/dstack [hunk 7]
```diff
@@ -40,17 +41,47 @@ obj_map = {
...
+
+          # Set all defaults
+          $logger.debug(File.basename(__FILE__)) { "Setting defaults for #{type} object #{k}" }
+          obj_map[type][k].set_defaults!
+      end
+    end
+
+    # Deployment Type: Canary
+    if rel['meta']['deployment']['type'].include?("canary")
+      puts "Deployment Type Canary is not supported yet!".red
```
Comment (mhktp) [inline exe/dstack]: This should probably be `$logger.error`

File: exe/dstack [hunk 8]
```diff
@@ -63,77 +94,86 @@ obj_map.each do |type, obj|
...
-$logger.debug(File.basename(__FILE__)) { "Starting dstack engine" }
+$logger.debug(File.basename(__FILE__)) { 'Starting dstack engine' }
 case $opts[:command]
   when 'encrypt'
     $logger.debug(File.basename(__FILE__)) { 'Encrypting declaration file(s)' }
     release.encrypt
     $logger.info('encrypt') { 'Encrypted config files. Please verify before committing.' }
   else
-    imakecmd   = ($opts[:command] == 'deploy') ? 'create' : $opts[:command]
+    imakecmd   = ($opts[:command] == 'create') ? 'create' : $opts[:command]
```
Comment (mhktp) [inline exe/dstack]: I think we've discussed this before, but we must update `imake` to support `create` before merging this PR.

File: lib/dstack/typedefs/container.rb [hunk 9]
```diff
@@ -27,6 +27,27 @@ def schema
       'required'             => ['docker_image', 'internal_port'], # Specify required settings
       'additionalProperties' => false, # This should always be defined - it restricts input to what is defined in the schema,
       'properties'           => {
+        'access' => {
+          'type' => 'string',
+          "pattern": "^(External|Internal)$"
```
Comment (mhktp) [inline lib/dstack/typedefs/container.rb]: ```External/Internal``` should probably be lowercase here to consistent with the rest of the yaml style

File: lib/dstack/typedefs/container.rb [hunk 10]
```diff
@@ -450,7 +624,7 @@ def generate_routing_config
     keyhash['domains'][domain]                        = {}
     keyhash['domains'][domain]['landingpage']         = self['routing']['landingpage'] if self['routing'].key? 'landingpage'
     keyhash['domains'][domain]['forcessl']            = self['routing']['forcessl'] if self['routing'].key? 'forcessl'
-    keyhash['domains'][domain]['nginx/optimizations'] = self['nginx_optimizations'] if self['routing'].key? 'nginx_optimizations'
+    keyhash['domains'][domain]['nginx/optimizations'] = self['routing']['nginx_optimizations'] if self['routing'].key? 'nginx_optimizations'
```
Comment (mhktp) [inline lib/dstack/typedefs/container.rb]: We should push another KV here with the ELB scheme (internal/external).

File: lib/dstack/typedefs/container.rb (hunk 9 above)
Comment (rdrabo) [inline lib/dstack/typedefs/container.rb]: Fixed.

File: lib/dstack/options.rb (hunk 3 above)
Comment (rdrabo) [inline lib/dstack/options.rb]: Fixed

File: lib/dstack/typedefs/container.rb (hunk 10 above)
Comment (rdrabo) [inline lib/dstack/typedefs/container.rb]: Done

**Pull request merged**